mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java

```diff
@@ -88,30 +91,28 @@ public void removeGlobalListener(MailboxListener listener, MailboxSession sessio
 
     @Override
     public void event(Event event) {
-        deliverEventToGlobalListeners(event);
-        if (event instanceof MailboxEvent) {
-            mailboxEvent((MailboxEvent) event);
-        }
-    }
-
-    private void mailboxEvent(MailboxEvent mailboxEvent) {
-        Collection<MailboxListener> listenerSnapshot = registry.getLocalMailboxListeners(mailboxEvent.getMailboxId());
-        if (mailboxEvent instanceof MailboxDeletion && listenerSnapshot.size() > 0) {
-            registry.deleteRegistryFor(mailboxEvent.getMailboxId());
+        ImmutableList<MailboxListener> listeners = ImmutableList.<MailboxListener>builder()
+            .addAll(registry.getGlobalListeners())
```

Comment:
we don't need to optimize it because it's probably very cheap but the right datastructure for this kind of situation would be "persistent datastructure"